src/seedu/addressbook/Main.java

```diff
@@ -106,12 +107,19 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        CommandResult result = null;
         try {
             command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+            result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }
+        catch (StorageWriteToReadOnlyException e)
+        {
+            ui.showToUser();
```

Comment:
What is shown to the user? Recall the purpose of catching exceptions, it allows you to gracefully "exit" from an operation in the event of an error.
It'll be good if there is some meaningful operations or messages shown to the user.